clang/lib/Sema/SemaChecking.cpp

```diff
@@ -2210,6 +2222,9 @@ Sema::CheckBuiltinFunctionCall(FunctionDecl *FDecl, unsigned BuiltinID,
   case Builtin::BI__builtin_alloca_uninitialized:
     Diag(TheCall->getBeginLoc(), diag::warn_alloca)
         << TheCall->getDirectCallee();
+    if (getLangOpts().OpenCL) {
+      builtinAllocaAddrSpace(*this, TheCall);
+    }
```

Comment:
A bit late tot this review, but right now we have `getASTAllocaAddressSpace()` in CodeGen/, could this be used moved so we can make use of it in Sema instead?